specs/CustomDownload.md

```diff
@@ -0,0 +1,775 @@
+                    m_settings->put_ShouldDisplayDefaultDownloadDialog(FALSE));
+
+                wil::com_ptr<ICoreWebView2Download> download;
+                CHECK_FAILURE(args->Download(&download));
+
+                UINT64 downloadSizeInBytes = 0;
+                CHECK_FAILURE(download->get_DownloadSizeInBytes(&downloadSizeInBytes));
+
+                wil::unique_cotaskmem_string uri;
+                CHECK_FAILURE(download->get_Uri(&uri));
+
+                wil::unique_cotaskmem_string mimeType;
+                CHECK_FAILURE(download->get_MimeType(&mimeType));
+
+                wil::unique_cotaskmem_string contentDisposition;
+                CHECK_FAILURE(download->get_ContentDisposition(&contentDisposition));
+
+                // Get the suggested path from the event args.
+                wil::unique_cotaskmem_string resultFilePath;
+                CHECK_FAILURE(args->get_ResultFilePath(&resultFilePath));
+
+                std::wstring prompt =
+                    std::wstring(L"Enter result file path or select `OK` to use default path. ") +
+                    L"Select `Cancel` to cancel the download.";
+
+                std::wstring description = std::wstring(L"URI: ") + uri.get() + L"\r\n" +
+                                            L"Mime type: " + mimeType.get() + L"\r\n" +
+                                            L"Download size in bytes: " + std::to_wstring(downloadSizeInBytes) +
+                                            L"\r\n";
+
+                TextInputDialog dialog(
+                    m_appWindow->GetMainWindow(), L"Download Starting", prompt.c_str(),
+                    description.c_str(), resultFilePath.get());
+                if (dialog.confirmed)
+                {
+                    // If user selects `OK`, the download will complete normally.
+                    // Result file path will be updated if a new one was provided.
+                    CHECK_FAILURE(args->put_ResultFilePath(dialog.input.c_str()));
+                    UpdateProgress(download.get());
+                }
+                else
+                {
+                    // If user selects `Cancel`, the download will be canceled.
+                    CHECK_FAILURE(args->put_Cancel(TRUE));
+                }
+            };
+
+            wil::com_ptr<ICoreWebView2Deferral> deferral;
+            CHECK_FAILURE(args->GetDeferral(&deferral));
+
+            // This function can be called to show the download dialog and
+            // complete the event at a later time, allowing the developer to
+            // perform async work before the event completes.
+            m_completeDeferredDownloadEvent = [showDialog, deferral] {
+                showDialog();
+                CHECK_FAILURE(deferral->Complete());
+            };
+
+            return S_OK;
+        })
+    .Get(),
+&m_downloadStartingToken));
+
+// Update download progress
+// Register a handler for the `DownloadProgressSizeInBytesChanged` event.
```

Comment:
Is this supposed to be the UpdateProgress method used above? Or where does this download variable come from? Is this part of the above sample snippet, or is this its own sample snippet? If so, you'd want to have download starting event handler that does this work below. If not, then good to hook it up to the above with appropriate method call or whatever

Reply:
Yes it's meant to be the UpdateProgress method

Reply:
Yes